src/kontrol/options.py

```diff
@@ -28,6 +28,7 @@ class ProveOptions:
     fail_fast: bool
     reinit: bool
     use_gas: bool
+    summary: dict | None
```

Comment:
Will this eventually be transformed into a `Deploymentsummary`? If so, consider doing it right in `__init__` (or even earlier), as the sooner you get rid of data that's not validated / whose type is not statically known the better,

Reply:
No, the `DeploymentSummary` is used only in the functionality of the `foundry_summary` function to generate the Solidity contract based on the `summary` dict.
But I get the idea and I can process the `summary` into a list of `SummaryEntry`s.

Reply:
Now, `summary` is `list[SummaryEntry] | None`.